tools/integration/src/index.ts

```diff
@@ -758,126 +774,318 @@ async function handleExport(argv: any) {
+        if (logger.isDebugEnabled()) {
+            logger.debug(`Response data: \n${JSON.stringify(response.data)}`);
         }
 
-        logger.info(`Total dashboard(s) processed: ${sanitizedIdObjects.length}`)
+        return response.data;
+    } catch (error) {
+        handleAxiosError(error, `event (id=${eventId})`);
+        return null;
     }
 }
 
-function printDirectoryTree(dirPath: string, rootLabel: string, indent: string = ''): void {
-    const isRoot = indent === '';
-    if (isRoot) {
-      logger.info(rootLabel);
+async function getEventList(server: string, token: string, axiosInstance: any): Promise<any[]> {
+    try {
+        const url = `https://${server}/api/events/settings/event-specifications/custom`;
+        logger.info(`Getting event list from ${url}`);
+
+        const response = await axiosInstance.get(url, {
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': `apiToken ${token}`
+            }
+        });
+
+        logger.info(`Successfully got event list: ${response.status}`);
+        if (logger.isDebugEnabled()) {
+            logger.debug(`Response data: \n${JSON.stringify(response.data)}`);
+        }
+
+        return response.data;
+    } catch (error) {
+        handleAxiosError(error, `event list`);
+        return [];
     }
+}
 
-    const files = fs.readdirSync(dirPath);
-    const lastIndex = files.length - 1;
+function saveEvent(dir: string, id: string, name: string, event: any) {
+    try {
+        const filename = `${name}.json`;
+        const filepath = path.join(dir, filename);
+        logger.info(`Saving event (id=${id}) to ${filepath}`);
+        fs.writeFileSync(filepath, JSON.stringify(event, null, 2));
+        logger.info(`Event (id=${id}) saved successfully`);
+    } catch (error) {
+        logger.error(`Error saving event (id=${id}):`, error);
+    }
+}
 
-    files.forEach((file, index) => {
-      const fullPath = path.join(dirPath, file);
-      const isDirectory = fs.statSync(fullPath).isDirectory();
-      const isLast = index === lastIndex;
-      const prefix = isLast ? '└── ' : '├── ';
+function filterEventsBy(idObjects: any[], include: string[]): any[] {
+    return idObjects.filter(obj => {
+        return include.every(condition => {
+            const [key, rawValue] = condition.split('=');
+            const value = rawValue?.replace(/^"|"$/g, '');
 
-      logger.info(indent + prefix + file);
+            if (key === 'name' || key === 'title') {
+                return new RegExp(value, 'i').test(obj.name ?? '');
+            } else if (key === 'id') {
+                return obj.id === value;
+            }
 
-      if (isDirectory) {
-        const newIndent = indent + (isLast ? '    ' : '│   ');
-        printDirectoryTree(fullPath, rootLabel, newIndent);
-      }
+            return false;
+        });
+    });
+}
+
+function sanitizeEventTitles(idObjects: any[]): any[] {
```

Comment:
Shall we combine the sanitize logic for both event title and dashboard title, because they are very similar.